nix/codex.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    types mkEnableOption mkOption mkIf literalExpression;
+  
+  toml = pkgs.formats.toml { };
+
+  cfg = config.services.codex;
+in
+{
+  options = {
+    services.codex = {
+      enable = mkEnableOption "Codex Node service.";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.callPackage ./default.nix { };
+        defaultText = literalExpression "pkgs.codex";
+        description = lib.mdDoc "Package to use as Codex node.";
+      };
+
+      configFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Path to the Codex configuration file.";
+      };
+
+      configObject = lib.mkOption {
```

Comment:
The usual name for this is `settings`. See https://search.nixos.org/options for examples.